tests/components/zwave_js/test_init.py

```diff
@@ -290,3 +292,41 @@ async def test_remove_entry(hass, stop_addon, uninstall_addon, caplog):
     assert entry.state == ENTRY_STATE_NOT_LOADED
     assert len(hass.config_entries.async_entries(DOMAIN)) == 0
     assert "Failed to uninstall the Z-Wave JS add-on" in caplog.text
+
+
+async def test_removed_device(hass, client, multiple_devices, integration):
+    """Test that the device registry gets updated when a device gets removed."""
+    nodes = multiple_devices
+
+    # Verify how many nodes are available
+    assert len(client.driver.controller.nodes) == 2
+
+    # Make sure there are the same number of devices
+    dev_reg = await device_registry.async_get_registry(hass)
+    device_entries = device_registry.async_entries_for_config_entry(
+        dev_reg, integration.entry_id
+    )
+    assert len(device_entries) == 2
+
+    # Check how many entities there are
+    ent_reg = await entity_registry.async_get_registry(hass)
+    entity_entries = entity_registry.async_entries_for_config_entry(
+        ent_reg, integration.entry_id
+    )
+    assert len(entity_entries) == 18
+
+    # Remove a node and reload the entry
+    old_node = nodes.pop(13)
+    await hass.config_entries.async_reload(integration.entry_id)
+
```

Comment:
Add a block till done here to make sure the entity entries are removed before we check.

Reply:
```suggestion
    await hass.async_block_till_done()

```